conformance/lang/expressions/shift-expression/right_shift.balt

```diff
@@ -0,0 +1,4537 @@
+Labels: int:Signed16, int:Signed32, int:Signed8, int:Unsigned16, int:Unsigned32, int:Unsigned8,
+        nil-literal, optional-type, shift-expr, unary-minus, union-type, value:toBalString
+
+function init() {
+    int:Unsigned32 x = 4294967285;
+    int:Signed8? a = 3;
+    int:Signed8? b = -123;
+    int:Unsigned8? c = 3;
+    int:Signed16? d = 2;
+    int:Signed16? e = -32758;
+    int:Unsigned16? f = 5;
+    int:Signed32? g = 3;
+    int:Signed32? h = -2147483641;
+    int:Unsigned32? i = 1;
+    int:Unsigned8? j = ();
+    int:Unsigned8? k = ();
+    int:Signed16? l = ();
+    int:Unsigned16? m = ();
+    int:Signed32? n = ();
+    int:Unsigned32? o = ();
+
+    int:Unsigned32|() x1 = x >> a;
+    io:println(x1); // @output 536870910
+    int:Unsigned32?|float x2 = x >> b;
+    io:println(x2); // @output 134217727
+
+    int:Unsigned32? x3 = x >> c;
+    io:println(x3); // @output 536870910
+
+    int:Unsigned32? x4 = x >> d;
+    io:println(x4); // @output 1073741821
+    int:Unsigned32? x5 = x >> e;
+    io:println(x5); // @output 4194303
+
+    int:Unsigned32? x6 = x >> f;
+    io:println(x6); // @output 134217727
+
+    int:Unsigned32? x7 = x >> g;
+    io:println(x7); // @output 536870910
+    int:Unsigned32? x8 = x >> h;
+    io:println(x8); // @output 33554431
+
+    int:Unsigned32|() x9 = x >> i;
+    io:println(x9); // @output 2147483642
+
+    int:Unsigned32|() x11 = x >> j;
+    io:println(x11.toBalString()); // @output ()
+    int:Unsigned32|() x12 = x >> k;
+    io:println(x12.toBalString()); // @output ()
+    int:Unsigned32|() x13 = x >> l;
+    io:println(x13.toBalString()); // @output ()
+    int:Unsigned32|() x14 = x >> m;
+    io:println(x14.toBalString()); // @output ()
+    int:Unsigned32|() x15 = x >> n;
+    io:println(x15.toBalString()); // @output ()
+    int:Unsigned32|() x16 = x >> o;
+    io:println(x16.toBalString()); // @output ()
+}
+
+Test-Case: error
+Description: Test resultant type of the '>>' expression with 'int:Unsigned32' as the LHS operand and optional builtin
+             subtypes of integer as the RHS operand.
+Labels: decimal, float, int, int:Signed16, int:Signed8, int:Unsigned16, int:Unsigned32, int:Unsigned8, optional-type,
+        shift-expr
+
+function errorFunction(int:Unsigned32 a, int:Signed8? b, int:Unsigned8? c, int:Signed16? d, int:Unsigned16? e, int:Signed32? f,
+                         int:Unsigned32? g) {
+    int:Unsigned8 _ = a >> b; // @error expected 'int:Unsigned8' but found 'int:Unsigned32?'
+    float _ = a >> c; // @error expected 'float' but found 'int:Unsigned32?'
+    int:Unsigned8 _ = a >> d; // @error expected 'int:Unsigned8' but found 'int:Unsigned32?'
+    int:Signed16 _ = a >> e; // @error expected 'int:Signed16' but found 'int:Unsigned32?'
+    decimal _ = a >> f; // @error expected 'decimal' but found 'int:Unsigned32?'
+    int:Unsigned16 _ = a >> g; // @error expected 'int:Unsigned16' but found 'int:Unsigned32?'
+}
```

Comment:
We are testing the statement ` if left hand operand is a subtype of int:UnsignedK when K is 8, 16 or 32, then the static type of the result is int:UnsignedN where N is the smallest such K;` here right? It's better to update the description with that information.

Reply:
Updated in https://github.com/ballerina-platform/ballerina-spec/pull/953/commits/2fe94debf0c3d87472283ccb4c9b99ae3142eeb9.